### PR TITLE
tools: support more update flow in pd-heartbeat-bench

### LINKED ISSUE #5743: tools: pd-heartbeat-bench is unexpected about update flow
## Bug Report

1. `update flow` add value rather than set value, which makes flow increase always.
26336c94-fff2-4100-9f9f-0b4b3a611748
90678386-3874-4f20-9589-2565b8d081b7
image

2.  `pd-heartbeat-bench` only update region info by region heartbeat, which makes read info is empty
627760eb-7685-4441-8798-54b5cd623eaf

3. `pd-heartbeat-bench` doesn't update query info.
image
90678386-3874-4f20-9589-2565b8d081b7

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #5743

### What is changed and how does it work?
1. support hot peer stat in store heartbeat
2. support size sum in store heartbeat
3. support correct update flow 
4. support query info in heartbeat
image
image
image
image

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-heartbeat-bench/main.go [hunk 1]
```diff
@@ -371,39 +418,54 @@ func main() {
...
+				storesStats := regions.collectStoresStats(cfg.StoreCount)
+				for i := 0; i < cfg.StoreCount; i++ {
+					wg.Add(1)
+					storeStat := storesStats[i]
+					go func() {
+						defer wg.Done()
+						cli.StoreHeartbeat(ctx, &pdpb.StoreHeartbeatRequest{Header: header(), Stats: storeStat})
+					}()
+				}
+				wg.Wait()
```
Comment (rleungx) [inline tools/pd-heartbeat-bench/main.go:442]: The region heartbeat and store heartbeat can happen at the same time, but after this PR, the region heartbeat will wait until all stores have finished heartbeat.

File: tools/pd-heartbeat-bench/main.go [hunk 2]
```diff
@@ -257,21 +248,33 @@ func (rs *Regions) update(replica int) {
...
+		if region.ApproximateSize == bytesUnit {
+			region.ApproximateSize = bytesUnit * 2
+		} else {
+			region.ApproximateSize = bytesUnit
+		}
+		if region.ApproximateKeys == keysUint {
+			region.ApproximateKeys = keysUint * 2
+		} else {
+			region.ApproximateKeys = keysUint
+		}
```
Comment (rleungx) [inline tools/pd-heartbeat-bench/main.go:260]: Will it prevent the same region from updating twice?

File: tools/pd-heartbeat-bench/main.go [hunk 3]
```diff
@@ -371,39 +410,54 @@ func main() {
...
+			if storeUpdateRound%6 == 0 {
+				rep := newReport(cfg)
+				r := rep.Stats()
+				startTime := time.Now()
+				for i := 1; i <= cfg.StoreCount; i++ {
+					id := uint64(i)
+					wg.Add(1)
+					go regions.handleRegionHeartbeat(wg, streams[id], id, rep)
+				}
+				wg.Wait()
```
Comment (rleungx) [inline tools/pd-heartbeat-bench/main.go:444]: The store heartbeat still needs to wait for the region heartbeat to finish.

Comment (nolouch) [reply tools/pd-heartbeat-bench/main.go:444]: cc @lhy1024

File: tools/pd-heartbeat-bench/main.go [hunk 4]
```diff
@@ -366,12 +430,12 @@ func main() {
 	regions := new(Regions)
 	regions.init(cfg)
 	log.Info("finish init regions")
-
+	go handleStoreHeartbeat(ctx, cfg, cli, regions)
```
[the remark this answers is not in the document]
Comment (lhy1024) [reply tools/pd-heartbeat-bench/main.go:433]: yes, but it will soon be back to normal

**Review by HunDunDM — APPROVED** [review]

File: tools/pd-heartbeat-bench/main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rleungx) [reply tools/pd-heartbeat-bench/main.go:433]: Why not use the original way?

[the remark this answers is not in the document]
Comment (lhy1024) [reply tools/pd-heartbeat-bench/main.go:433]: we need to report hot read region info in the store heartbeat.

**Comment by nolouch** [comment]
ptal @rleungx

File: tools/pd-heartbeat-bench/main.go [hunk 5]
```diff
@@ -322,6 +327,50 @@ func (rs *Regions) handleRegionHeartbeat(wg *sync.WaitGroup, stream pdpb.PD_Regi
 	log.Info("store finish one round region heartbeat", zap.Uint64("store-id", storeID), zap.Duration("cost-time", time.Since(start)))
 }
 
+func (rs *Regions) collectStoresStats(storeCount int) []*pdpb.StoreStats {
+	stores := make([]*pdpb.StoreStats, storeCount)
+	now := uint64(time.Now().Unix())
+	for i := 0; i < storeCount; i++ {
```
Comment (rleungx) [inline tools/pd-heartbeat-bench/main.go:333]: How about `for i := 1; i <= storeCount; i++`? We can avoid -1.

**Review by rleungx — APPROVED** [review]
The rest LGTM.

**Review by rleungx — APPROVED** [review]
The rest LGTM.

**Pull request merged**